thirdeye/thirdeye-frontend/app/pods/rootcause/controller.js

```diff
@@ -283,6 +312,92 @@ export default Ember.Controller.extend({
     onHeatmapClick([dimension, subdimension]) {
       // TODO: do something with the call back
       console.log('heatmap click registerd for: ', dimension, subdimension);
+    },
+
+    /**
+     * Updates the filters cache with newly selected filters
+     * @param {String} eventType - event type (i.e. Holiday, GCN)
+     * @param {Object} subFilter - representation of subfilter and its values (i.e. {key: "country", value: ["US"]})
+     */
+    updateFiltersCache(eventType, subFilter) {
+      // Initialize objects before assignment in the cache
+      if (!this.filtersCache[eventType]) {
+        this.filtersCache[eventType] = {};
+      }
+      if (!this.filtersCache[eventType].subFilters) {
+        this.filtersCache[eventType].subFilters = {};
+      }
+
+      if (subFilter && !this.filtersCache[eventType].subFilters[subFilter.key]) {
+        this.filtersCache[eventType].subFilters[subFilter.key] = [];
+      }
+
+      // Set the subFilters value based on event type
+      if (subFilter) {
+        this.filtersCache[eventType].subFilters[subFilter.key] = subFilter.value;
+      }
+    },
+
+    /**
+     * Updates the urns cache by event type
+     * @param {String} eventType
+     * @param {Array} urns - array of urns
+     */
+    updateUrnsCache(eventType, urns) {
+      this.urnsCache[eventType] = urns;
+    },
+
+    /**
+     * Filters urns based on event type, subfilters selected, and user text from filter search bar
+     * @param {String} eventType - type of event
+     * @param {Object} subFilter - filter within event (i.e. country, region for Holiday)
+     * @param {String} text - user's input from filter search bar
+     */
+    filterUrns(eventType, subFilter, text) {
```

Comment:
This is ok for poc, but let's refactor it together once you're back.

Reply:
It's not very clear what the whole function does. We can improve readability by breaking it down into smaller utils/helper functions